apps/challenges/models.py

```diff
@@ -80,6 +84,13 @@ def is_active(self):
         return False
 
 
+@receiver(post_save, sender=Challenge)
+def publish_message_on_change(instance, created=False, **kwargs):
```

Comment:
Yeah, I actually wanted to get rid of this too. `post_save` signal provides `update_fields` and that could have done the task, but for some reasons, `update_fields` is `None` when `post_save` signal is invoked. Am looking into that and will follow up asap.